core/config/config_helper.hpp

```diff
@@ -200,7 +200,9 @@ get_value(const pnode& config)
  * This is specialization for floating point type
  */
 template <typename ValueType>
-inline std::enable_if_t<std::is_floating_point<ValueType>::value, ValueType>
+inline std::enable_if_t<std::is_floating_point<ValueType>::value ||
```

Comment:
I think this structure will be useful everywhere. Maybe it would be nice to have our own `is_floating_point` which includes `half` ? So all you will need to do is `gko::is_floating_point<ValueType>::value` ?

Reply:
I only use it here because others usually need to be writen separately